src/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -438,6 +451,13 @@ function discoverController($scope, config, courier, $route, $window, Notifier,
           if (!sortFn) hit.$$_counted = true;
 
           const fields = _.keys(indexPattern.flattenHit(hit));
+
+          // since flattenHit removed the _source field, but we want user's to be able to select it,
+          // we're manually adding it back
+          if (hit._source) {
+            fields.push('_source');
```

Comment:
Even though the `_source` field is now present in the field chooser list, it can not be added, because the "add" button is hidden [in the template](https://github.com/kobelb/kibana/blob/038f6cae28687cfdbb2aa215cff77b0174e3f9c5/src/core_plugins/kibana/public/discover/components/field_chooser/discover_field.html#L17).